packages/hash/frontend/src/components/EntityEditor/EntityFieldLinkEditor.tsx

```diff
@@ -173,28 +174,39 @@ export const EntityFieldLinkEditor: VoidFunctionComponent<
         }
       />
       {allowsMultipleSelections
-        ? linksOnField.map(({ link, linkedEntity }) => (
-            <div className={tw`flex my-6`} key={link.linkId}>
-              <div className={tw`font-bold w-32`}>
-                {/*
-                 * @todo remove the need for this component to know about links and accountId
-                 *    e.g. pass a GoToEntity component into it
-                 */}
-                <Link
-                  href={`/${linkedEntity.accountId}/entities/${linkedEntity.entityId}`}
+        ? linksOnField
+            .filter(
+              (
+                linkOnField,
+              ): linkOnField is {
+                link: BlockProtocolLink & SingleTargetLinkFields;
+                linkedEntity: BlockProtocolEntity;
+              } => {
+                return "linkId" in linkOnField.link;
```

Comment:
You have this "linkId" in x type guard kinda all over the place. Can you abstract it to a function that is something lie `isSingleLink`? Even if you end up having to duplicate that function (as `Media.tsx` can't really import it) its still clearer what's going on if its named.

Reply:
I've tried to replace as many of these checks as I could into a helper function here: https://github.com/hashintel/hash/pull/413/commits/127b95580ab18ddd3da3cf6e729097fdfa4116f4